src/controllers/addressBook/addressBook.ts

```diff
@@ -0,0 +1,160 @@
+import { getAddress } from 'ethers'
+
+import { Storage } from '../../interfaces/storage'
+import EventEmitter from '../eventEmitter/eventEmitter'
+
+export type Contacts = Array<{
+  name: string
+  address: string
+  isWalletAccount?: boolean
+}>
+
+/**
+ * Address Book controller
+ */
+
+export class AddressBookController extends EventEmitter {
+  // Manually added contact (stored in storage)
+  #manuallyAddedContacts: Contacts = []
+
+  // Contacts, generated on the fly from the accounts in the wallet (not stored in storage)
+  #accountsInWalletContacts: Contacts = []
+
+  #storage: Storage
+
+  constructor(storage: Storage) {
+    super()
+
+    this.#storage = storage
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.#load()
```

Comment:
Please apply the `this.#initialLoadPromise = this.#load()` pattern that we aim to follow for the new controllers. You can take a look how this is implemented for the KeystoreController:

```
// Holds the initial load promise, so that one can wait until it completes
#initialLoadPromise: Promise<void>
```